versioned/memory/src/main/java/org/projectnessie/versioned/memory/InMemoryVersionStore.java

```diff
@@ -515,13 +560,40 @@ public Collector collectGarbage() {
     return InactiveCollector.of();
   }
 
+  @Override
+  public Stream<WithTypeAndId<Key, EnumT>> getGlobalKeys() {
+    synchronized (lock) {
```

Comment:
I think it might be good to replace `ConcurrentHashMap` with a `NonBlockingHashMap` (https://github.com/JCTools/JCTools/blob/master/jctools-core/src/main/java/org/jctools/maps/NonBlockingHashMap.java#L83) and remove the `lock`

Reply:
right now the lock is there to ensure that this hashmap isn't used to read during a commit. rather than using a nice collection I chose the lock because it has to synchronise across 3 different hashmaps during the commit. I haven't decided if we really need to lock this method at all (it might be that concurrently reading form this map while a commit is happenig is ok). But the lock is definitely needed (imho) for the commit method. Do you disagree?

Reply:
Better not mix CHM + synchronized/locks. It may give people a false sense of safety to access the CHM outside the lock. But that's rather irrelevant as long it's a prototype ;)

Reply:
agreed, the `ConcurrentHashMap` is only there because i am lazy. These shoudl all be a `HashMap`s